python/foglamp/services/core/api/task.py

```diff
@@ -0,0 +1,244 @@
+        count = await check_schedules(storage, name)
+        if count != 0:
+            raise web.HTTPBadRequest(reason='A schedule with that name already exists')
+
+        # Now first create the scheduled process entry for the new task
+        cmdln_params = [', "--{}={}"'.format(i, v) for i, v in cmd_params.items()] if cmd_params is not None and len(cmd_params) > 0 else []
+        cmdln_params_str = "".join(cmdln_params)
+        script = '["tasks/{}"{}]'.format(task_type, cmdln_params_str)
+        payload = PayloadBuilder().INSERT(name=name, script=script).payload()
+        try:
+            res = await storage.insert_into_tbl("scheduled_processes", payload)
+        except StorageServerError as ex:
+            err_response = ex.error
+            raise web.HTTPInternalServerError(reason='Failed to created scheduled process. {}'.format(err_response))
+        except Exception as ins_ex:
+            raise web.HTTPInternalServerError(reason='Failed to created scheduled process. {}'.format(str(ins_ex)))
+
+        # If successful then create a configuration entry from plugin configuration
+        try:
+            # Create a configuration category from the configuration defined in the plugin
+            category_desc = plugin_config['plugin']['description']
+            config_mgr = ConfigurationManager(storage)
+            await config_mgr.create_category(category_name=name,
+                                             category_description=category_desc,
+                                             category_value=plugin_config,
+                                             keep_original_items=True)
+        except Exception as ex:
+            await revert_scheduled_processes(storage, plugin)  # Revert scheduled_process entry
+            raise web.HTTPInternalServerError(reason='Failed to create plugin configuration. {}'.format(str(ex)))
+
+        # If all successful then lastly add a schedule to run the new task at startup
+        try:
+            schedule = TimedSchedule() if schedule_type == Schedule.Type.TIMED else \
+                       IntervalSchedule() if schedule_type == Schedule.Type.INTERVAL else \
+                       ManualSchedule()
+            schedule.name = name
+            schedule.process_name = name
+            schedule.day = schedule_day
+            m, s = divmod(schedule_time if schedule_time is not None else 0, 60)
+            h, m = divmod(m, 60)
+            schedule.time = datetime.time().replace(hour=h, minute=m, second=s)
+            schedule.repeat = datetime.timedelta(schedule_repeat if schedule_repeat is not None else 0)
+            schedule.exclusive = True
+            schedule.enabled = False  # if "enabled" is supplied, it gets activated in save_schedule() via is_enabled flag
+
+            # Save schedule
+            await server.Server.scheduler.save_schedule(schedule, is_enabled)
+            schedule = await server.Server.scheduler.get_schedule_by_name(name)
```

Comment:
does not `save_schedule` return the schedule_id ?

Reply:
Nope